[14.0][FIX] account_financial_report +  account_tax_balance: Fix tests

joao-p-marques opened the pull request:
On multi-company settings, we need to make sure we are searching and creating objects with the correct company associated.
Also take into account previously created account groups

@Tecnativa
TT28423

joao-p-marques committed:
[FIX] account_financial_report: Fix tests

On multi-company settings, we need to make sure we are searching and creating objects with the correct company associated.
Also take into account previously created account groups

 TT28423

pedrobaeza commented:
I'm not sure why you need to restrict all the searches. You only need to make sure the user is in and only in the right company. ORM record rules should do the rest.

joao-p-marques committed:
[FIX] account_financial_report: Refactor tests

Use core account test base, helping to ensure we have the correct accounting environment setup.
If any of the localization modules are loaded, this will make this tests be skipped, instead of simply failing

TT28423

joao-p-marques committed:
[FIX] account_tax_balance: Fix tests

Take into account previously created taxes

TT28423

joao-p-marques commented:
> I'm not sure why you need to restrict all the searches. You only need to make sure the user is in and only in the right company. ORM record rules should do the rest.

Yes, but with the latest refactor I added, using the base tests from the `account` module, 2 companies are created associated to the same test user. So, to avoid problems, it is better to restrict the search to a single one.

joao-p-marques commented:
Tests were failing now because of `account_tax_balance` due to an unrelated problem. I pushed another commit to fix it.

pedrobaeza commented:
/ocabot merge patch
